ci/scripts/r_install_system_dependencies.sh

```diff
@@ -37,10 +37,10 @@ if [ "$ARROW_S3" == "ON" ] || [ "$ARROW_GCS" == "ON" ] || [ "$ARROW_R_DEV" == "T
   # Install curl and OpenSSL for S3/GCS support
   case "$PACKAGE_MANAGER" in
     apt-get)
-      apt-get install -y libcurl4-openssl-dev libssl-dev
+      apt-get install -y libcurl4-openssl-dev libssl-dev pkg-config
```

Comment:
How common is it to be on ubuntu and not have pkg-config installed? Does this mean we would require pkg-config in order for the binary to be used?

Reply:
> How common is it to be on ubuntu and not have pkg-config installed?

Hmm. I'm not sure preference of R users on Ubuntu but I think that users who have GCC also have pkg-config.

But `rocker/r-ver:4.0.0` and `rocker/r-ver:3.6.3` doesn't have pkg-config. So we need this.

> Does this mean we would require pkg-config in order for the binary to be used?

Yes.

We can fallback to source build when a user doesn't have pkg-config:

```diff
diff --git a/r/configure b/r/configure
index ff6a9dacc4..08fa58d327 100755
--- a/r/configure
+++ b/r/configure
@@ -176,6 +176,7 @@ else
       if [ "${ARROW_DEPENDENCY_SOURCE}" = "AUTO" ] && \
            [ "${PKG_CONFIG_AVAILABLE}" = "false" ]; then
         export ARROW_DEPENDENCY_SOURCE=BUNDLED
+        export LIBARROW_BINARY=false
         echo "**** pkg-config not installed, setting ARROW_DEPENDENCY_SOURCE=BUNDLED"
       fi
 
```

Do you want me to add this change to this PR?

Reply:
I'd rather ship a binary that just works. Would it be terrible to grep for the CXX11_ABI in arrow.pc and add it to `PKG_CFLAGS`?

If we were to set `LIBARROW_BINARY=false` if no pkg-config, it should be in its own `"${PKG_CONFIG_AVAILABLE}" = "false"` check, not connected to whatever the value of `ARROW_DEPENDENCY_SOURCE` is since that's about building libarrow from source.

Reply:
> I'd rather ship a binary that just works. Would it be terrible to grep for the CXX11_ABI in arrow.pc and add it to `PKG_CFLAGS`?

It will work. I've added the logic but I'm afraid of increasing ad-hoc logics again.

> If we were to set `LIBARROW_BINARY=false` if no pkg-config, it should be in its own `"${PKG_CONFIG_AVAILABLE}" = "false"` check, not connected to whatever the value of `ARROW_DEPENDENCY_SOURCE` is since that's about building libarrow from source.

It makes sense.